homeassistant/core.py

```diff
@@ -308,6 +300,18 @@ def async_stop(self) -> None:
 
         self.loop.stop()
 
+    @callback
+    def async_stop_handler(self, *args):
+        """Stop Home Assistant."""
+        self.exit_code = 0
+        self.loop.create_task(self.async_stop())
+
+    @callback
+    def async_restart_handler(self, *args):
```

Comment:
We should rename this to `async_restart`:

```python
def async_restart(self):
    self.exit_code = RESTART_EXIT_CODE
    return self.async_stop()
```

And if we default exit_code to 0, async_stop doesn't have to assign anything.